components/splitter/splitter.component.ts

```diff
@@ -0,0 +1,382 @@
+  readonly nzResize = output<number[]>();
+  readonly nzResizeEnd = output<number[]>();
+
+  readonly destroy$ = inject(NzDestroyService);
+  readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+
+  /** ------------------- Panels ------------------- */
+  // Get all panels from content children
+  readonly panels = contentChildren(NzSplitterPanelComponent);
+  // Subscribe to the change of properties
+  readonly panelProps = computed(() =>
+    this.panels().map(panel => ({
+      defaultSize: panel.nzDefaultSize(),
+      size: panel.nzSize(),
+      min: panel.nzMin(),
+      max: panel.nzMax(),
+      resizable: panel.nzResizable(),
+      collapsible: coerceCollapsible(panel.nzCollapsible()),
+      contentTemplate: panel.contentTemplate()
+    }))
+  );
+
+  /** ------------------- Sizes ------------------- */
+  /**
+   * The size of the container, used to calculate the percentage size and flex basis of each panel.
+   */
+  readonly containerSize = computed(() =>
+    this.nzLayout() === 'horizontal'
+      ? this.elementRef.nativeElement.clientWidth || 0
+      : this.elementRef.nativeElement.clientHeight || 0
+  );
+  readonly innerSizes = linkedSignal({
+    source: this.panelProps,
+    computation: source => source.map(panel => panel.defaultSize)
+  });
+  readonly sizes = computed(() => {
+    let emptyCount = 0;
+    const containerSize = this.containerSize();
+    const innerSizes = this.innerSizes();
+    const sizes = this.panelProps().map((panel, index) => {
+      const innerSize = innerSizes[index];
+      const size = panel.size ?? innerSize;
+      const hasSize = panel.size !== undefined;
+
+      // Calculate the percentage size of each panel.
+      let percentage: number | undefined;
+      if (isPercent(size)) {
+        percentage = getPercentValue(size);
+      } else if (size || size === 0) {
+        const num = Number(size);
+        if (!isNaN(num)) {
+          percentage = num / containerSize;
+        }
+      } else {
+        percentage = undefined;
+        emptyCount++;
+      }
+
+      // Calculate the min and max percentage size of each panel.
+      const minSize = panel.min;
+      const maxSize = panel.max;
+      const postPercentMinSize = isPercent(minSize) ? getPercentValue(minSize) : (minSize || 0) / containerSize;
+      const postPercentMaxSize = isPercent(maxSize)
+        ? getPercentValue(maxSize)
+        : (maxSize || containerSize) / containerSize;
+
+      return {
+        innerSize,
+        size,
+        hasSize,
+        percentage,
+        min: minSize,
+        max: maxSize,
+        postPercentMinSize,
+        postPercentMaxSize
+      } as PanelSize;
+    });
+
+    const totalPercentage = sizes.reduce((acc, { percentage }) => acc + (percentage ?? 0), 0);
+
+    if (totalPercentage > 1 && !emptyCount) {
+      // If total percentage is larger than 1, we will scale it down.
+      const scale = 1 / totalPercentage;
+      sizes.forEach(size => {
```

Comment:
`forEach` looks weird when modifying mutable element's properties.  
I think an indexed `for` loop would be clearer, and you could also extract the code snippet into its own function.